src/graphics/Screen.cpp

```diff
@@ -2150,13 +2151,23 @@ void Screen::setFrames(bool holdPosition = false)
 
     setFastFramerate(); // Draw ASAP
 
+    // In some situations, we'd like to return to the same frame
+    static size_t oldNumFrames = numframes;
     if (holdPosition) {
-        ui->switchToFrame(currentFrameNum); // Attempt to return to same frame after rebuilding the frames,
-        // if holdPosition is true (currently only Screen::handleStatusUpdate calls this
-    } else {
-        continue; // We leave the displayed frame as it is or chnage focuse to new frame
-    }   
-}    
+        if (oldFrame == (oldNumFrames - 1))   // If we were on the final frame (settings)
+            ui->switchToFrame(numframes - 1); // then move back to the final frame
+
+        else if (oldFrame == (oldNumFrames - 2)) // If we were on the log buffer frame
+            ui->switchToFrame(numframes - 2);    // then move back there
+
+        else if (oldFrame > numframes - 1) // If we were on a frame that no longer exists
+            ui->switchToFrame(0);          // back to the first frame
+
+        else
+            ui->switchToFrame(oldFrame); // Otherwise, go back to the same frame
+    }
+    oldNumFrames = numframes; // Store how many frames we have, in case we want to "restore position" next time
+}
```

Comment:
That logic is clear now .
Will flash all to my T-Echo tonight and test.
(b^_^)b